src/firmware_logger_device.cpp

```diff
@@ -44,91 +66,151 @@ namespace librealsense
+                                            fw_logs::fw_log_data * parsed_msg )
+    {
+        bool result = false;
+        if( _parser && parsed_msg && fw_log_msg )
         {
-            fw_logs::fw_logs_binary_data data;
-            data = _flash_logs.front();
-            _flash_logs.pop();
-            binary_data = data;
+            *parsed_msg = _parser->parse_fw_log( fw_log_msg );
             result = true;
         }
+
         return result;
     }
 
-    bool firmware_logger_device::init_parser(std::string xml_content)
+    legacy_firmware_logger_device::legacy_firmware_logger_device( std::shared_ptr< const device_info > const & dev_info,
+                                                                  std::shared_ptr< hw_monitor > hardware_monitor,
+                                                                  const command & fw_logs_command,
+                                                                  const command & flash_logs_command )
+        : firmware_logger_device( dev_info, hardware_monitor, fw_logs_command )
+        , _flash_logs_command( flash_logs_command )
+        , _flash_logs()
+        , _flash_logs_initialized( false )
     {
-        _parser = new fw_logs::fw_logs_parser(xml_content);
+    }
+
+    bool legacy_firmware_logger_device::init_parser( std::string xml_content )
+    {
+        _parser = std::make_unique< fw_logs::legacy_fw_logs_parser >( xml_content );
 
-        return (_parser != nullptr);
+        return ( _parser != nullptr );
     }
 
-    bool firmware_logger_device::parse_log(const fw_logs::fw_logs_binary_data* fw_log_msg,
-        fw_logs::fw_log_data* parsed_msg)
+    size_t legacy_firmware_logger_device::get_log_size( uint8_t * buff ) const
+    {
+        // Log size constant, no need to call parser like firmware_logger_device::get_log_size.
+        return sizeof( fw_logs::legacy_fw_log_binary );
+    }
+
+    command legacy_firmware_logger_device::get_update_command()
+    {
+        return _fw_logs_command;
+    }
+    
+    bool legacy_firmware_logger_device::get_flash_log( fw_logs::fw_logs_binary_data & binary_data )
     {
         bool result = false;
-        if (_parser && parsed_msg && fw_log_msg)
+
+        if( ! _flash_logs_initialized )
+        {
+            get_flash_logs_from_hw_monitor();
+        }
+
+        if( ! _flash_logs.empty() )
         {
-            *parsed_msg = _parser->parse_fw_log(fw_log_msg);
+            binary_data = std::move( _flash_logs.front() );
+            _flash_logs.pop();
             result = true;
         }
 
         return result;
     }
+    
+    void legacy_firmware_logger_device::get_flash_logs_from_hw_monitor()
+    {
+        auto res = _hw_monitor->send( _flash_logs_command );
+
+        if( res.empty() )
+        {
+            LOG_INFO( "Getting Flash logs failed!" );
+            return;
+        }
+
+        // Erasing header
+        int size_of_flash_logs_header = 27;
```

Comment:
Please make this a const int data member of the class

Reply:
This is legacy code I did not change, but I will update this to be `constexpr`.
I prefer not to make it a member in the h file, as it is relevant only here.